### DISCUSSION TITLE
defaultValues and dirty

### DISCUSSION THREAD

**Discussion by fde312** [react-hook-form/react-hook-form#7654] [Q&A] [answered]
Is there a way to make the ``defaultValues`` ``dirty`` with ``reset``? I would need the prefilled inputs to be in ``dirtyFields`` of ``formState``?

I know that you can do that also with ``setValue``, but I want to set it before the inputs are ``registered``

Thank you! :)

**Comment by Moshyfawn** [comment]
What about [the `keepDirty: true` prop](https://react-hook-form.com/api/useform/reset) for the reset form method?
See [this CSB](https://codesandbox.io/s/old-paper-9gir9?file=/src/App.tsx)

**Reply by fde312** [reply]
In your CSB, the ``ìsDiry`` state changes only after a manual input. 

What I need is to have ``ìsDirty=true`` after the ``defaultValues`` are set, so without any user interaction.

**Comment by Moshyfawn** [comment]
If you only need the form state to be set to dirty _(I'd imaging, you want something like "enable submit button on form `isDirty`")_, you can do
```js
useEffect(() => {
  setValue("name", "", { shouldDirty: true });
  reset(defaultValues, { keepDirty: true });
}, [setValue, reset]);
```
See [this CSB](https://codesandbox.io/s/funny-sinoussi-bppbv?file=/src/App.tsx)

I don't think there's gonna be another "proper" way to do what you want, as `formState.isDirty` is RHF comparing the current form field values against their defaults.

**Comment by bluebill1049** [comment, marked as answer]
Thanks @Moshyfawn for the help. I think you can use `keepDefaultValues: true`, so the `defaultValues` will be remained and dirty check will be checked against new values and existing `defaultValeus`.

**Reply by Moshyfawn** [reply]
Ah, that's genius lol!
```js
const { reset } = useForm({
  defaultValues: {
    name: ""
  }
})

useEffect(() => {
  reset({ name: "Bob" }, { keepDefaultValues: true })
}, [reset])
```

[CSB example](https://codesandbox.io/s/green-firefly-oemyb?file=/src/App.tsx)

**Reply by fde312** [reply]
that actually worked for me, thank you so much! :)
